pkg/splunk/controller/statefulset.go

```diff
@@ -139,6 +264,17 @@ func UpdateStatefulSetPods(c splcommon.ControllerClient, statefulSet *appsv1.Sta
 			return splcommon.PhaseError, err
 		}
 
+		// Check if the revision was updated successfully for Statefulset.
+		// It so can happen that it may take few seconds for the update to be
+		// reflected in the resource. In that case, just return from here and
+		// check the status back in the next reconcile loop.
+		var ret bool
+		ret, err = checkAndUpdatePodRevision(c, statefulSet, readyReplicas-1, skipRecheckUpdate)
+		if !ret || err != nil {
```

Comment:
Can we move this check outside of the scale up/scale down checks since its common for both.  Though I guess the `readyReplicas` is different by 1.

Reply:
Yes readyReplicas was different in both cases hence did not move it out.